### ISSUE TITLE
INVALID_QUERY when browsing collections using archive field

### ISSUE THREAD

**Issue by licitdev** [directus/directus#8096]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

When opening a collection, an unexpected error occurs.
<img width="408" alt="image" src="https://user-images.githubusercontent.com/26413686/133717155-149baece-f346-408e-b839-e42416ca2d84.png">

This issue only surfaces for **existing** collections using a `Status` or `Archive` field.
<img width="716" alt="image" src="https://user-images.githubusercontent.com/26413686/133719081-64d3dc7b-9735-47c5-a90c-94ef4e3f6d6b.png">

### To Reproduce

Browse a collection. 

Not sure where the issue is but the url that causes this error is `.../items/something?limit=25&fields[]=id&fields[]=name&sort=id&page=1&filter=%7B"status":%7B"_neq":null%7D%7D`.

Upon clicking the reset page preferences button, the request url becomes `/items/something?limit=25&fields[]=id&fields[]=name&fields[]=status&sort=id&page=1&filter=%7B%7D` and loads fine.

However, when browsing to another collection and back, the same error occurs.

### Tested workarounds
- If the archive field is deselected for **some** collections, the error vanishes.
- The error can also be removed temporarily by setting an advanced filter then removing it.
  - A preset is automatically saved. However, when the preset is deleted, the error comes back.
  - Remember to refresh the entire webpage after deleting the preset, to see error resurface.

### What version of Directus are you using?

v9.0.0-rc.93

### What version of Node.js are you using?

v16

### What database are you using?

Postgres

### What browser are you using?

Safari

### What operating system are you using?

macOS

### How are you deploying Directus?

PaaS

**Comment by azrikahar** [comment]
Thanks for the detailed info! Is the `Status` field the default one available when you create this collection? or did you add it manually afterwards? I'm wondering why there's no value for `Archive Value` and `Unarchive Value` as seen in your screenshot.

With that said, would you mind trying to set Archive Value and Unarchive Value to `archived` and `draft` respectively, and see does the issue goes away? of course assuming you are using the default status field. Sample screenshot:

![chrome_9SRNYNhAlc](https://user-images.githubusercontent.com/42867097/133728176-abc60ac5-4e68-4eb5-907e-33d03f379d0d.png)

**Comment by licitdev** [comment]
@azrikahar Oh wow yes! Thank you! That fixed the issue. It didn't have to be set previously. 
Should the archive value field be made mandatory if an archived field was set or perhaps the archive value could default to `archived`?

**Comment by azrikahar** [comment]
@licitdev glad that solves it. Hmm honestly I can't say I entirely recall that it didn't have to be set previously, but I just tested in rc.92 and rc.93 just now, and ticking/choosing Status when creating a new Collection does defaults Archive Value and Unarchive Value to `archived` and `draft` automatically. Can you try creating a new dummy collection and see whether that is true for your setup as well?

**Comment by licitdev** [comment]
@azrikahar Yes it was automatically added but I've removed it quite sometime back as the value differs. I did not notice any errors until I've updated it to v9.0.0-rc.93. So I am not sure which RC did the issue surfaced.

**Comment by azrikahar** [comment]
@licitdev do you know/recall what was the previous version you're on before going to rc.93?

**Comment by azrikahar** [comment]
@licitdev ah.. apparently this isn't really an issue with archive at all, as you mentioned it worked for you previously even if it's null.

Apparently the recent support for geometry & map necessitate deep parsing of query filters, but in the process, it changed `_neq: null` into `_neq: {}` (empty object), which is why your error shows it must be string/number/boolean (null is also ok), but it ended up being an empty object instead.

So basically this is a general query filter issue, not isolated to archive filter only, but your scenario uncovered this quirk 😆 Will create the PR soon

**Comment by licitdev** [comment]
@azrikahar Wow you're really fast! 👍🏻 Thanks again for the quick fix that solves the collection browsing glitch~

**Comment by azrikahar** [comment]
@licitdev welcome, but then again good bug report precedes fast bug fix! 

though I'll still need confirmation on whether my fix is appropriate, so let's wait and see.
